src/util.c

```diff
@@ -288,7 +288,11 @@ size_t execute_system_cmd_by_process(char *command, char *type, char **output)
 
 	count = getline(output, &len, pfp);
 
-	fclose(pfp);
+	if (pclose(pfp) == -1) {
+		PRINT_ERR("Error closing process process");
```

Comment:
```suggestion
		perror("pclose");
```

Please include the error details from `errno` in the message.

Reply:
The error message is not optimal. But using `PRINT_ERR()` alignes with previous error logging. `perror()` is not used once in this project.

Reply:
I see what you mean but in my opinion the error message is useless if it doesn't include the reason. We can revisit this PR once the existing error handling has been converted to include the reason given by the OS.

Reply:
IMHO it would be better to modify `PRINT_ERR()` to log `errno` / `strerror(errno)` instead of replacing it with `perror()`. What do you think?

Reply:
Yes, I haven't checked that all the call sites myself though. I also think the function should be changed to lowercase since it is not a macro.

Reply:
True. Dunno why it's uppercase. But I initially thought it's a macro as well.